src/pyhf/constraints.py

```diff
@@ -129,58 +160,79 @@ def _precompute(self):
                     default_backend.shape(self.par_indices[parslice])
                 )
 
-        if poisson_constraint_rate_indices:
-            poisson_rate_idc = default_backend.concatenate(
-                list(
-                    map(
-                        lambda x: default_backend.astensor(x, dtype='int'),
-                        poisson_constraint_rate_indices,
-                    )
-                )
+        if self.parameter_helper.index_selection:
+            poisson_rate_fac = default_backend.astensor(
+                default_backend.concatenate(poisson_constraint_rate_factors),
+                dtype='float',
             )
-            poisson_rate_fac = default_backend.concatenate(
-                list(
-                    map(
-                        lambda x: default_backend.astensor(x, dtype='float'),
-                        poisson_constraint_rate_factors,
-                    )
-                )
-            )
-            poisson_data = default_backend.concatenate(
-                list(
-                    map(
-                        lambda x: default_backend.astensor(x, dtype='int'),
-                        poisson_constraint_data,
-                    )
-                )
+            poisson_data = default_backend.astensor(
+                default_backend.concatenate(poisson_constraint_data), dtype='int'
             )
 
-            self.poisson_data = tensorlib.astensor(
+            self._poisson_data = default_backend.astensor(
```

Comment:
not sure what this line is doing as poisson_data is already in the tensor of default_backend from previous line